src/Mvc/Mvc.ApiExplorer/src/DefaultApiDescriptionProvider.cs

```diff
@@ -258,9 +261,30 @@ private void ProcessRouteParameters(ApiParameterContext context)
                         parameter.Source = BindingSource.Path;
                     }
                 }
+                else
+                {
+                    if (parameter.Source == BindingSource.Path &&
+                        parameter.ModelMetadata is DefaultModelMetadata defaultModelMetadata &&
+                        !defaultModelMetadata.Attributes.Attributes.OfType<IFromRouteMetadata>().Any())
+                    {
+                        // If we didn't see the parameter in the route and no FromRoute metadata is set, it probably means
+                        // the parameter binding source was inferred (InferParameterBindingInfoConvention)  
+                        // probably because another route to this action contains it as route parameter.
+                        // let's add it to be removed from the API description
+                        inferredPathParametersToRemove.Add(parameter);
```

Comment:
Could we do this inline without the extra List allocation? Use a indexed loop that starts from the end so you can call `RemoveAt(i)`